tests/gdx-tests/src/com/badlogic/gdx/tests/g3d/shadows/utils/DirectionalAnalyzer.java

```diff
@@ -0,0 +1,34 @@
+/*******************************************************************************
+ * Copyright 2011 See AUTHORS file.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+package com.badlogic.gdx.tests.g3d.shadows.utils;
+
+import com.badlogic.gdx.graphics.Camera;
+import com.badlogic.gdx.graphics.g3d.environment.BaseLight;
+import com.badlogic.gdx.math.Frustum;
+import com.badlogic.gdx.math.Vector3;
+
+/** Directional Analyzer compute the camera's properties needed by a directional light
```

Comment:
Can you expand the class or method doc a bit? I guess this is intended to compute the camera properties, so that it sees (from the lights perspective) everything that casts shadows which are visible within the frustum. Ofcourse it doesnt have knowledge about the scene, so it might not include something large that is not visible to the camera but still could cast visible shadows. It looks like (from the comment) that it is restricted to a directional light (e.g. shouldnt be used for a spotlight).